runner/src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -595,6 +596,27 @@ export class PageControllerBase {
       }
     }
 
+    if (form_session_identifier) {
+      const clientFileFields = viewModel.components
+        .filter((component) => component.type === "ClientSideFileUploadField")
+        .map((component) => component.model);
+
+      const pageId = `${this.model.basePath}${this.pageDef.path}`;
+
+      if (clientFileFields.length > 0) {
+        //TODO temp fix, hardcoded to one a we only every allow 1 file upload on one page
+        let name = clientFileFields[0].name ?? "";
+        const folderPath = `${pageId}/${name}`;
+        const files = await uploadService.listFilesInBucketFolder(
+          `${form_session_identifier}/${folderPath}`,
+          form_session_identifier
+        );
+        if (files) {
+          payload[name] = files[0].Key;
+        }
+      }
+    }
```

Comment:
Does this happen when the user hits "Save and continue" - if so it's a bit tricky to track the state there since we're dealing with client side uploads, theoretically the user could go back via the summary page, change the file or delete their file (if optional) and the state wouldn't be updated automatically as these actions are entirely client-side, they'd have to hit "Save and continue" for the state to be updated, however everywhere else we use the state of the bucket itself to circumvent this.  

I think when the user finalises and submits their entire application, we should grab the current state of each client side file upload component, and update these fields in the database on the application-store, therefore we don't need to deal with tracking the state with "Save and continue" and can just use the state of the bucket.  

It is a bit tedious having to implement this in multiple places... as this is one place I obviously missed (the report when you submit) - but not sure how else we would do it with the client side upload nature, open to options though, what do you think?

Reply:
Relevant JIRA: https://digital.dclg.gov.uk/jira/browse/FS-2842